### PR TITLE
Added support to Karpenter provisioner

### PR DESCRIPTION
**Pull request by aeciopires**
#### What this PR does / why we need it:
- Added support to Karpenter provisioner
- Added new values
- Updated documentation
- Change version of chart
- Fixed issue #12 

#### Special notes for your reviewer:
- Tested only using EKS cluster 1.23 in AWS with Karpenter 0.19.2

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: charts/zabbix/README.md [hunk 1]
```diff
@@ -484,11 +488,12 @@ The following tables lists the configurable parameters of the chart and their de
 | zabbixweb.readinessProbe.timeoutSeconds | int | `5` | Number of seconds after which the probe times out |
 | zabbixweb.replicaCount | int | `1` | Number of replicas of ``zabbixweb`` module |
 | zabbixweb.resources | object | `{}` | Requests and limits of pod resources. See: [https://kubernetes.io/docs/concepts/configuration/manage-resources-containers](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers) |
-| zabbixweb.service | object | `{"annotations":{},"clusterIP":null,"port":80,"type":"ClusterIP"}` | Certificate containing certificates for SAML configuration saml_certs_secret_name: zabbix-web-samlcerts |
+| zabbixweb.service | object | `{"annotations":{},"clusterIP":null,"nodePort":31080,"port":80,"type":"NodePort"}` | Certificate containing certificates for SAML configuration saml_certs_secret_name: zabbix-web-samlcerts |
```
Comment (sa-ChristianAnton) [inline charts/zabbix/README.md:491]: Why change Web service to NodePort by default? Is this good practice? Personally, I don't want to have NodePort services by default. There are many scenarios where they are either not allowed to use or not desired. Ingress should be the way to go to publish zabbix-web in production, and NodePort or port-forward is for testing scenarios only, imo.

File: charts/zabbix/README.md [hunk 2]
```diff
@@ -484,11 +488,12 @@ The following tables lists the configurable parameters of the chart and their de
...
 | zabbixweb.service.annotations | object | `{}` | Annotations for the zabbix-web service |
 | zabbixweb.service.clusterIP | string | `nil` | Cluster IP for Zabbix web |
+| zabbixweb.service.nodePort | int | `31080` | NodePort port to allocate (only if service.type = NodePort) |
 | zabbixweb.service.port | int | `80` | Port to expose service |
-| zabbixweb.service.type | string | `"ClusterIP"` | Type of service for Zabbix web |
+| zabbixweb.service.type | string | `"NodePort"` | Type of service for Zabbix web |
```
Comment (sa-ChristianAnton) [inline charts/zabbix/README.md:496]: Same as above regarding NodePort service

File: charts/zabbix/docs/example/kind/values.yaml [hunk 3]
```diff
@@ -89,7 +89,7 @@ zabbixweb:
   ZBX_SERVER_HOST: zabbix-zabbix-server
   ZBX_SERVER_PORT: 10051
   service:
-    type: ClusterIP
+    type: NodePort
```
Comment (sa-ChristianAnton) [inline charts/zabbix/docs/example/kind/values.yaml:92]: Same as above regarding NodePort service

File: charts/zabbix/templates/statefulset-zabbix-proxy.yaml [hunk 4]
```diff
@@ -28,14 +28,20 @@ spec:
       {{- with .Values.zabbixproxy.extraPodSpecs }}
       {{- toYaml . | nindent 6 }}
       {{- end }}
-      {{- with .Values.nodeSelector }}
-      {{- with .Values.zabbixproxy.extraInitContainers }}
+       {{- with .Values.zabbixproxy.extraInitContainers }}
```
Comment (sa-ChristianAnton) [inline charts/zabbix/templates/statefulset-zabbix-proxy.yaml:31]: Indentation? One space too much before bracket?

File: charts/zabbix/values.yaml [hunk 5]
```diff
@@ -309,13 +309,13 @@ zabbixweb:
   #saml_certs_secret_name: zabbix-web-samlcerts
   service:
     # -- Type of service for Zabbix web
-    type: ClusterIP
+    type: NodePort
```
Comment (sa-ChristianAnton) [inline charts/zabbix/values.yaml:312]: Same as above regarding NodePort service

File: charts/zabbix/values.yaml [hunk 6]
```diff
@@ -309,13 +309,13 @@ zabbixweb:
...
     # -- Cluster IP for Zabbix web
     clusterIP:
     # -- Port to expose service
     port: 80
     # -- NodePort port to allocate (only if service.type = NodePort)
-    #nodePort: 31080
+    nodePort: 31080
```
Comment (sa-ChristianAnton) [inline charts/zabbix/values.yaml:318]: Same as above regarding NodePort service

File: charts/zabbix/README.md (hunk 1 above)
Comment (aeciopires) [reply charts/zabbix/README.md:491]: Hello @sa-ChristianAnton!

To use ingress, the service must be NodePort. I switched to make it easier. But I understand your concerns and have rolled back.

File: charts/zabbix/README.md (hunk 2 above)
Comment (aeciopires) [reply charts/zabbix/README.md:496]: Hi @sa-ChristianAnton!

To use ingress, the service must be NodePort. I switched to make it easier. But I understand your concerns and have rolled back.

File: charts/zabbix/docs/example/kind/values.yaml (hunk 3 above)
Comment (aeciopires) [reply charts/zabbix/docs/example/kind/values.yaml:92]: Hi @sa-ChristianAnton!

To use ingress, the service must be NodePort. I switched to make it easier. But I understand your concerns and have rolled back.

File: charts/zabbix/values.yaml (hunk 6 above)
Comment (aeciopires) [reply charts/zabbix/values.yaml:318]: Hi @sa-ChristianAnton!

To use ingress, the service must be NodePort. I switched to make it easier. But I understand your concerns and have rolled back.

File: charts/zabbix/values.yaml (hunk 5 above)
Comment (aeciopires) [reply charts/zabbix/values.yaml]: Hi @sa-ChristianAnton!

To use ingress, the service must be NodePort. I switched to make it easier. But I understand your concerns and have rolled back.

File: charts/zabbix/templates/statefulset-zabbix-proxy.yaml (hunk 4 above)
Comment (aeciopires) [reply charts/zabbix/templates/statefulset-zabbix-proxy.yaml]: Fixed. Thanks!

**Review by sa-ChristianAnton — APPROVED** [review]

**Pull request merged**